### DISCUSSION TITLE
Game suffers from bad performance on fresh Arch (btw) install, and Vulkan doesn't work at all

### DISCUSSION THREAD

**Discussion by StalkyBoi43815** [ppy/osu#28501] [Q&A] [answered]
Basically title. Game sits at ~300fps in menu and has noticeable input lag. Also, when I try to use Vulkan, the next time I start the game, it says something along the lines of "Vulkan could not initialize, reverting to default renderer". 

New Arch install, NoVidia card, KDE Plasma (X11)
[compressed-logs.zip](https://github.com/user-attachments/files/15858762/compressed-logs.zip)

**Comment by Stoppedpuma** [comment]
> ~300fps in menu and has noticeable input lag.

Performance optimisation is still a work in progress but I'm pretty you should be getting this bad of performance. This is likely an issue on your system. Regarding input lag, X11 isn't really known for being low latency. I would check to make sure the compositor is disabled, otherwise this is probably related to the performance you're seeing.

> Vulkan doesn't work

`ln -s /usr/lib/libdl.so.2 /usr/lib/libdl.so` should fix this issue and allow you to use Vulkan, otherwise find the correct library names with `ldconfig -p | grep libdl`

**Reply by StalkyBoi43815** [reply]
Performance fixed itself but now whenever I start a map on Vulkan the game's display freezes. Sound and inputs still work.

**Reply by Stoppedpuma** [reply, marked as answer]
Vulkan is experimental and has very mixed results on different systems, you can report about your experiences at #27659.

**Reply by StalkyBoi43815** [reply]
> Vulkan is experimental and has very mixed results on different systems, you can report about your experiences at #27659.

actually good answer

**Comment by airi019** [comment]
i complained about this here a while ago and the solution is like some linux player said, to remove pipewire pulseaudio and anything else and use pure alsa with a window manager like i3 or dwm. 

Thanks to that you wont be able to use kde or any other desktop environment and obviously wont be able to play audio from other apps while running osu. 

After doing that i get decent latency with -30ms for offset. 

When using pipewire seems like it has its own thing for managing latency so offset on osu is all over the place and I had to remove it.

Also, I mounted ext4 with noatime option and that more or less improves performance a lot and reduce lags and freezes.
